Add image support in CLI interface

MaxRcd opened the pull request:
Hello,

I worked on adding image support to CLI
Image size is taken from the image file itself

Because we can't really set the opt.centered = true flag from CLI. We use the tag {%%image} in documents instead.

Regards,
Maxime

edi9999 commented:
Hello @MaxRcd ,

thanks for the pull request.

However, I'm concerned about two points : 

1. **Security**, and this one is quite important to me... If we do `return fs.readFileSync(tagValue, "binary");` , this means that if the user has control over the tagValues, he could be reading anyfile from Disk (including all js files, or also some other files). I don't know whether we can find a solution for this, since the image files could come from somewhere else.

2. **Configuration** : Rather than having the image module autoload if present, I would prefer to make it configurable (something like config.imageModule). This way, we don't do a try catch, and the user controls himself whether he wants to use the imageModule (without having to uninstall/install the image-module)

MaxRcd commented:
Hi @edi9999,

1. I could add file extension checking (.png / .jpg / .jpeg / .bmp / .gif). Do you think this is enough ? We also could check the file mime type however it means adding another npm package as a dependency which is not justified in my opinion but I'm OK with this solution too.

Just figured out that all images are assumed to be PNG files.
```
//docxtemplater-image-module file index.js
getNextImageName() {
		const name = `image_generated_${this.imageNumber}.png`;
		this.imageNumber++;
		return name;
	}
```

2. I agree

Regards,
Maxime

edi9999 commented:
For 1), I find the approach of extension checking to be non optimal.

I think it would be even better if the user gave us a path to a folder (for example /var/screenshots), and we would allow only to have files coming from that folder, no matter the name of the file or its extension

edi9999 commented:
For 2), instead of having a key for imageModule specifically, I think we can do : 

```
modules: ['docxtemplater-image-module']
```

This way, we can add other modules as well without changing the interface each time
(the name of the npm package)

MaxRcd commented:
Hi @edi9999,

I updated the CLI

1) Users can specify _config.imageDir_ parameter in _config.json_. This a path that could be either absolute or relative to the current working directory of the process.

2) Agree with you. I implemented it this way

Typical _config.json_ file:
```
{
  "config.modules": ["docxtemplater-image-module"],
  "config.imageDir": "myDir",
  "config.inputFile": "examples/tagImage.pptx",
  "config.outputFile": "output.pptx",
  "config.debug": true,
  "first_name": "John",
  "last_name": "Smith",
  "age": 62,
  "image": "examples/images/image-tagging.png"
}
```

edi9999 commented:
Thanks!
